src/box/vinyl.c

```diff
@@ -3102,6 +3102,10 @@ vinyl_engine_join(struct engine *engine, void *arg, struct xstream *stream)
 			if (vy_join_send_tuple(stream, it->lsm->space_id,
 					       entry.stmt) != 0)
 				return -1;
+			if (fiber_is_cancelled()) {
+				diag_set(FiberIsCancelled);
+				return -1;
+			}
```

Comment:
This looks redundant - `vy_join_send_tuple` calls `relay_send_initial_join_row`, which is a cancellation point.

Reply:
It is not yet a cancellation point. But after discussion f2f we decided to make `coio_write_timeout` a cancellation point. So this change is dropped.